### DISCUSSION TITLE
How to manage cas.pem at DIRAC server?

### DISCUSSION THREAD

**Discussion by hmiyake** [DIRACGrid/DIRAC#7826] [Q&A] [answered]
Hello, 

I realized sometimes following files are made a lot in /tmp of DIRAC server
```
/tmp/cas.dzz8welw.pem
```
Curiously it is a lot in some DIRAC server
```
$ ls -1 /tmp/cas*.pem|wc
    322     322    7084
```
while none found in the other one...

I suppose this file is created by generateCAFile() of  DIRAC/Core/Security/Utilities.py but don't understand how to prevent to create those files in /tmp.

Do we need to keep cas.pem updated anywhere (say /opt/dirac/etc/grid-security/cas.pem) in DIRAC server?
If so how can what is correct procedure?
I found we had genAllCAs.sh in older DIRAC release but it is not shipped with DIRAC v8, if I'm correct.

My worry is our DIRAC deployment including update is lack of necessary procedure to set/update that file...

Or if we don't need such file, how can we avoid many files creation in /tmp ?

**Comment by fstagni** [comment]
Hello, these are invoked by BundleDelivery. Would it make you any problem if you increase the frequency of `/tmp` file cleanup (`tmpfiles.d` config I think?

**Reply by hmiyake** [reply, marked as answer]
Thank you for the answer and suggestion! 
Honestly I hesitate to increase cleanup frequency of whole tmp files...tried to find how to clean specific files by tmpfiles.d but could not...

Finally I prepared cron script using find (`find /tmp/cas*.pem -atime +3 -delete`)

For my education, recent DIRAC does not produce /opt/dirac/etc/grid-security/cas.pem anymore?
